week2/homework/src/index.js

```diff
@@ -1,3 +1,125 @@
 'use strict';
 
 // TODO: Write the homework code in this file
+const fs = require('fs');
+const fileName = './index.txt';
+const defaultEncoding = 'UTF8';
+
+const program = require('commander');
+
+program
+    .version('0.0.1');
+program
+    .command('list')
+    .description('Shows current to-dos')
+    .action(() => listItems());
+program
+    .command('add <index>')
+    .description('add a item to the list')
+    .action(index => addItem(index));
+program
+    .command('remove <index>')
+    .description('remove a item from list')
+    .action(index => removeItem(index));
+program
+    .command('reset')
+    .description('removes all the items from the list')
+    .action(() => resetList());
+program
+    .command('update <index> <newItem>')
+    .description('update a item from the list')
+    .action((index, updatedTask) => updateItem(index, updatedTask));
+program
+    .command('help')
+    .description('show help')
+    .action(() => printHelp());
+program.parse(process.argv);
+
+function listItems() {
+    fs.readFile(fileName, defaultEncoding, (err, data) => {
+        if (err) {
+            console.log(err);
+        }
+        else {
+            console.log(data);
+        }
+    });
+}
+
+function addItem(index) {
+    index = process.argv[3];
```

Comment:
This function already receives the correct `index` from `commander`, you don't need to compute it yourself. You can remove this line and the app still works correctly :)
The same is valid for all other functions. `commander` parses `process.argv` for you so you don't have to deal with this error-prone stuff 🎉